manager/container_test.go

```diff
@@ -226,3 +226,23 @@ func TestUpdateNvidiaStats(t *testing.T) {
 	assert.NotNil(t, err)
 	assert.Equal(t, info.ContainerStats{}, stats)
 }
+
+func TestOnDemandHousekeeping(t *testing.T) {
+	cd, _, _ := newTestContainerData(t)
+	statsList := itest.GenerateRandomStats(1, 4, 1*time.Second)
+	stats := statsList[0]
+
+	cd, mockHandler, memoryCache := newTestContainerData(t)
+	mockHandler.On("GetStats").Return(
+		stats,
+		nil,
+	)
+	go func() {
+		cd.housekeeping()
+	}()
+	cd.OnDemandHousekeeping(0 * time.Second)
```

Comment:
I would like to see a few other cases: non-zero call get's stats if necessary, but not if unnecessary, nil call does not get new stats, and multiple simultaneous calls only get stats once, and all return. You might need to plumb a fakeclock through to make this non-flaky.